### ISSUE TITLE
Is \drummode supported?

### ISSUE THREAD

**Issue by snappizz** [Abjad/abjad#773]

**Comment by josiah-wolf-oberholtzer** [comment]
tl;dr It is _somewhat_ supported. Could use some love.

Output works:

```
>>> from abjad import *
>>> drum_mode = Container()
>>> attach(LilyPondCommand('drummode', 'before'), drum_mode)
>>> drum_mode.extend('hihat4 hh bassdrum bd')
>>> drum_staff = Staff([drum_mode], context_name='DrumStaff')
>>> print(format(drum_staff))
\new DrumStaff {
    \drummode
    {
        hihat4
        hihat4
        bassdrum4
        bassdrum4
    }
}
```

Input doesn't work:

```
>>> Staff(r'\drummode { hihat4 }', context_name='DrumStaff')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/joberholtzer/Development/abjad/abjad/tools/scoretools/Staff.py", line 52, in __init__
    name=name,
  File "/Users/joberholtzer/Development/abjad/abjad/tools/scoretools/Context.py", line 57, in __init__
    name=name,
  File "/Users/joberholtzer/Development/abjad/abjad/tools/scoretools/Container.py", line 77, in __init__
    self._initialize_music(music)
  File "/Users/joberholtzer/Development/abjad/abjad/tools/scoretools/Container.py", line 740, in _initialize_music
    parsed = self._parse_string(music)
  File "/Users/joberholtzer/Development/abjad/abjad/tools/scoretools/Container.py", line 789, in _parse_string
    parsed = parse(user_input)
  File "/Users/joberholtzer/Development/abjad/abjad/tools/topleveltools/parse.py", line 47, in parse
    return _lilypond_parsers_by_language[language](arg)
  File "/Users/joberholtzer/Development/abjad/abjad/tools/lilypondparsertools/LilyPondParser.py", line 248, in __call__
    lexer=self._lexer,
  File "/Users/joberholtzer/Development/abjad/abjad/tools/lilypondparsertools/_parse.py", line 67, in _parse
    self.lookahead = get_token()     # Get the next token
  File "/Users/joberholtzer/.virtualenvs/supriya/lib/python3.5/site-packages/ply/lex.py", line 350, in token
    newtok = func(tok)
  File "/Users/joberholtzer/Development/abjad/abjad/tools/lilypondparsertools/LilyPondLexicalDefinition.py", line 483, in t_notes_421
    t.type = self.scan_escaped_word(t)
  File "/Users/joberholtzer/Development/abjad/abjad/tools/lilypondparsertools/LilyPondLexicalDefinition.py", line 854, in scan_escaped_word
    raise Exception(message)
Exception: unknown escaped word: '\\drummode'.
```

Our LilyPond parser doesn't know how to parse _a significant_ chunk of LilyPond syntax. To be fair to myself - since I wrote it - LilyPond's grammar is huge.

One of the things the parser _does_ know is about a dictionary of drum-mode note-names, housed in `ly/drums.ly`. I believe I scraped that from LilyPond's source by hand. That dictionary allows you to instantiate notes and chords via drum-mode note-names rather than pitch-names. The resulting pitches or chords aggregate `DrumNoteHead` objects instead of the usual pitched `NoteHead` objects.

Let me know if you need more details, or want to turn this into a more specific feature request.

**Comment by snappizz** [comment]
Thanks. I'm content with the `LilyPondCommand` solution.
